### PR TITLE
chore(schematics): add proprietary migrations

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        } catch {
+            // noop
+        }
+
+        return () => {};
+    };
+}
+
+function replaceScopePackages(tree: Tree): void {
+    const fileSystem = getFileSystem(tree);
```
Comment (vladimirpotekhin) [inline projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts:45]: getFileSystem (setActiveProject / saveActiveProject) is an expensive operation.
Why do we need call it (twice)? can we use filesystem from `main`?

File: projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                '@taiga-ui/proprietary-tds-icons',
+                '@taiga-ui/proprietary-tds-palette',
+            ].forEach(name => removePackageJsonDependency(tree, name));
+
+            addPackageJsonDependency(tree, {
+                name: '@taiga-ui/proprietary',
+                version: TUI_VERSION,
+            });
+
+            context.addTask(new NodePackageInstallTask());
```
Comment (vladimirpotekhin) [inline projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts:144]: Let's add this task `NodePackageInstallTask` in the bottom of main. We will need to remove the `experimental` package, so this task will always be useful, not only in proprietary projects

File: projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+function replaceAngularJsonStyles(options: TuiSchema, tree: Tree): Rule {
+    let proprietaryIcons: string | null = null;
+
+    try {
+        proprietaryIcons =
+            getPackageJsonDependency(tree, '@taiga-ui/proprietary-icons')?.version ||
+            getPackageJsonDependency(tree, '@taiga-ui/proprietary-tds-icons')?.version ||
+            null;
+
+        if (getSourceFiles('**/angular.json').length) {
```
Comment (vladimirpotekhin) [inline projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts:76]: Why do we need this check? what if the project doesn't have angular.json? (nx project with project.json for example)

File: projects/cdk/schematics/ng-update/v4/index.ts [hunk 4]
```diff
@@ -22,27 +22,32 @@ import {
...
+            migrateStyles();
+            showWarnings(context, MIGRATION_WARNINGS);
 
-        fileSystem.commitEdits();
-        saveActiveProject();
-    };
+            fileSystem.commitEdits();
+            saveActiveProject();
+        },
+        removeProprietaryPackages(),
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/schematics/ng-update/v4/index.ts]: fixed

File: projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts]: fixed

File: projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts (hunk 2 above)
Comment (splincode) [reply projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Comment by splincode** [comment]
@waterplea hello, please review me

**Review by waterplea — APPROVED** [review]

**Pull request merged**